zerver/lib/email_notifications.py

```diff
@@ -272,10 +275,16 @@ def message_header(message: Message) -> dict[str, Any]:
         elif message.recipient.type == Recipient.DIRECT_MESSAGE_GROUP:
             grouping = {"huddle": message.recipient_id}
             display_recipient = get_display_recipient(message.recipient)
-            narrow_link = direct_message_group_narrow_url(
-                user=user,
-                display_recipient=display_recipient,
-            )
+            if len(display_recipient) > 2:
+                narrow_link = direct_message_group_narrow_url(
+                    user=user,
+                    display_recipient=display_recipient,
+                )
+            else:
+                narrow_link = personal_narrow_url(
+                    realm=user.realm,
+                    sender=message.sender,
```

Comment:
If we're migrating to the concept that a 1:1 DM is just a special case of a `DirectMessageGroup` (namely, a `DirectMessageGroup` with just two participants), then wouldn't it make sense for this `if/else` to live inside of `direct_message_group_narrow_url` instead?

That is - `direct_message_group_narrow_url` should return the correct URL - possibly personal. Without the caller having to worry which function to call depending on `display_recipient` size?

Reply:
Good point. Updated.